### PR TITLE
Cleaned up the "-Wshorten-64-to-32" and "-Wsign-conversion" warnings.

### PR DESCRIPTION
**Pull request by kuzminrobin**

### CODE REVIEW

**Comment by kuzminrobin** [comment]
Please review.

File: src/Qir/Runtime/lib/QIR/callables.cpp [hunk 1]
```diff
@@ -20,10 +20,10 @@ using namespace Microsoft::Quantum;
 ==============================================================================*/
 extern "C"
 {
-    PTuple quantum__rt__tuple_create(int64_t size)
+    PTuple quantum__rt__tuple_create(int64_t size)  // TODO: Use unsigned integer type (breaking change).
```
Comment (ScottCarda-MS) [inline src/Qir/Runtime/lib/QIR/callables.cpp:23]: Do you need to take care of this ToDo before merging?

File: src/Qir/Runtime/lib/Simulators/FullstateSimulator.cpp [hunk 2]
```diff
@@ -99,7 +100,11 @@ namespace Quantum
         }
 
         const QUANTUM_SIMULATOR handle = 0;
-        unsigned simulatorId = -1;
+
+        using TSimulatorId = unsigned;      // TODO: Use `void*` or a fixed-size integer, starting in native simulator (breaking change).
```
Comment (ScottCarda-MS) [inline src/Qir/Runtime/lib/Simulators/FullstateSimulator.cpp:104]: Do you need to take care of this ToDo before merging?

File: src/Qir/Runtime/public/CoreTypes.hpp [hunk 3]
```diff
@@ -26,10 +26,10 @@
   dereferenced in client's code.
 ==============================================================================*/
 class QUBIT;
-typedef QUBIT* Qubit;
+typedef QUBIT* Qubit;   // TODO: Replace with `typedef uint64_t QubitId`. Remove all the `GetQubitId()`, `QUBIT`.
 
 class RESULT;
-typedef RESULT* Result;
+typedef RESULT* Result; // TODO: Replace with `typedef uintXX_t Result`, where XX is 8|16|32|64. Remove all the `RESULT`.
```
Comment (ScottCarda-MS) [inline src/Qir/Runtime/public/CoreTypes.hpp:32]: Do you need to take care of these ToDo's before merging?

**Review by ScottCarda-MS — APPROVED** [review]
I'm not so familiar with this area of the code, but nothing looked too scary 😄. There are a couple of ToDo's added in the code. Could you just do a quick double-check to make sure these aren't things that need to be addressed before merging in?

File: src/Qir/Runtime/lib/QIR/callables.cpp (hunk 1 above)
Comment (kuzminrobin) [reply src/Qir/Runtime/lib/QIR/callables.cpp:23]: I marked as ToDo the things that need a separate consideration (e.g. a synchronous change is needed in more than one repo or a careful change is required in multiple places in one commit).

File: src/Qir/Runtime/lib/Simulators/FullstateSimulator.cpp (hunk 2 above)
Comment (kuzminrobin) [reply src/Qir/Runtime/lib/Simulators/FullstateSimulator.cpp:104]: Resolved in a different discussion earlier.

File: src/Qir/Runtime/public/CoreTypes.hpp (hunk 3 above)
Comment (kuzminrobin) [reply src/Qir/Runtime/public/CoreTypes.hpp:32]: Resolved in a different discussion earlier.

**Pull request merged**